### PR TITLE
feature: kotlin/js and typescript

### PR DESCRIPTION
**Pull request by Ingwersaft**
PR adds support for Kotlin/JS target incl. Typescript types using the new Kotlin IR backend. Kotlin IR only exports limited types to Typescript which is the reason for the `AuthHandlerJS` wrapper.

TODO is the publishing to npm and some documentation how to consume it from Typescript

### CODE REVIEW

File: auth/build.gradle.kts [hunk 1]
```diff
@@ -55,6 +86,17 @@ kotlin {
                 implementation(Libs.ktorIOS)
             }
         }
+        val jsMain by getting {
+            dependencies {
+                implementation(Libs.ktorJS)
+                implementation(kotlin("test-js"))
```
Comment (gaebel) [inline auth/build.gradle.kts:92]: This should be in the jsTest target

File: README.md [hunk 2]
```diff
@@ -197,3 +202,15 @@ deleteUser(accessToken = "TOKEN_FROM_SIGN_IN_REQUEST"): Result<Unit>
 ## License
 
 Auth is available under the MIT license. See the LICENSE file for more info.
+
+## Contributing
```
Comment (gaebel) [inline README.md:206]: This should preceed License

File: auth/build.gradle.kts [hunk 3]
```diff
@@ -104,6 +146,66 @@ tasks {
     val iosX64Test by existing(KotlinNativeSimulatorTest::class) {
         filter.excludeTestsMatching("com.liftric.auth.AuthHandlerIntegrationTests")
     }
+
+    val testSecrets by creating(GetVaultSecretTask::class) {
+        secretPath.set("secret/apps/smartest/shared/cognito")
+    }
+
+    val createJsEnvHack by creating {
```
Comment (gaebel) [inline auth/build.gradle.kts:154]: We could use https://github.com/yshrsmz/BuildKonfig for this. But I'm fine with rolling this on our own.

Comment (Ingwersaft) [reply auth/build.gradle.kts:153]: For production code sure! But for getting two test variables into the test targets, I think this is good enough :)

**Comment by gaebel** [comment]
LGTM :) I've changed the workflow to also execute the js tests. Will merge when the pipeline succeeds.

**Review by gaebel — APPROVED** [review]

**Pull request merged**